lib/routers/task/index.js

```diff
@@ -58,6 +58,10 @@ router.use('/:taskId', async (req, res, next) => {
     const trainingCourseId = get(req.task, 'data.data.trainingCourseId');
     const trainingCourse = await req.models.TrainingCourse.query().findById(trainingCourseId);
 
+    if (!trainingCourse) {
+      return next(new BadRequestError('No associated training course found for personal licence'));
```

Comment:
Why a 400 here? What is the underlying reason that this error would hit?

Reply:
if the training course is somehow removed, but the task for the pil remains - I don't think it would happen in the wild, but it happened for me when I seeded one with out the other - it hangs as it's an uncaught promise rejection

Reply:
I was mostly curious as to why a 400 and not a 404 or 500.

It sounds like a 500 rather than a 400 if the issue is that we've somehow put the data into a weird state.

The reason this is important is that 4xx errors don't trigger alerts, so if something is knackered but returning 400s then we'll never know about it.

Reply:
a good point.

Reply:
You've changed a different instance, which I think was correct, but not this one.

Reply:
I think we all just need to quit. It's over.